cmd/thanos/store.go

```diff
@@ -206,9 +222,12 @@ func runStore(
 		return errors.Wrap(err, "create index cache")
 	}
 
+	garbageBlocksFinder := garbagecollector.NewGarbageBlocksFinder()
 	metaFetcher, err := block.NewMetaFetcher(logger, fetcherConcurrency, bkt, dataDir, extprom.WrapRegistererWithPrefix("thanos_", reg),
 		block.NewTimePartitionMetaFilter(filterConf.MinTime, filterConf.MaxTime).Filter,
 		block.NewLabelShardedMetaFilter(relabelConfig).Filter,
+		block.NewConsistencyDelayMetaFilter(logger, consistencyDelay).Filter,
+		garbagecollector.NewGarbageBlocksFilter(garbageBlocksFinder).Filter,
```

Comment:
:clap: :+1:

Reply:
Adding the garbageblocks filter is a good addition right?

Reply:
Yep! It should reduce RAM usage in certain cases.